framework/src/main/java/org/checkerframework/framework/stub/AnnotationFileElementTypes.java

```diff
@@ -873,4 +875,26 @@ private void prepJdkFromJar(URL resourceURL) {
             throw new BugInCF("Cannot open the jar file " + resourceURL.getFile(), e);
         }
     }
+
+    /**
+     * This method is invoked each time before {@link AnnotationFileParser} processes a top-level
+     * type.
+     *
+     * @param typeName The fully qualified name of the top-level type
+     */
+    void preProcessTopLevelType(String typeName) {
+        boolean success = processingClasses.add(typeName);
+        assert success;
```

Comment:
I'm also wondering whether to throw a `BugInCF` instead of an assert - if you're worried that this could arise in a client configuration, the assert is likely not enabled and a `BugInCF` would ensure we raise an error.

Reply:
I think `added` is more straightforward. I'm also using `BugInCF` now to ensure we always raise the error.